miniapps/nurbs/nurbs_ex5.cpp

```diff
@@ -0,0 +1,473 @@
+//               nurbs_ex3 -m ../../data/cube-nurbs.mesh
+
+
+#include "mfem.hpp"
+#include <fstream>
+#include <iostream>
+#include <algorithm>
+
+using namespace std;
+using namespace mfem;
+
+// Define the analytical solution and forcing terms / boundary conditions
+void uFun_ex(const Vector & x, Vector & u);
+double pFun_ex(const Vector & x);
+void fFun(const Vector & x, Vector & f);
+double gFun(const Vector & x);
+double f_natural(const Vector & x);
+
+int main(int argc, char *argv[])
+{
+   StopWatch chrono;
+
+   // 1. Parse command-line options.
+   const char *mesh_file = "../../data/square-nurbs.mesh";
+   int ref_levels = -1;
+   int order = 1;
+   bool pa = false;
+   const char *device_config = "cpu";
+   bool visualization = 1;
+
+   OptionsParser args(argc, argv);
+   args.AddOption(&mesh_file, "-m", "--mesh",
+                  "Mesh file to use.");
+   args.AddOption(&ref_levels, "-r", "--refine",
+                  "Number of times to refine the mesh uniformly, -1 for auto.");
+   args.AddOption(&order, "-o", "--order",
+                  "Finite element order (polynomial degree).");
+   args.AddOption(&pa, "-pa", "--partial-assembly", "-no-pa",
+                  "--no-partial-assembly", "Enable Partial Assembly.");
+   args.AddOption(&device_config, "-d", "--device",
+                  "Device configuration string, see Device::Configure().");
+   args.AddOption(&visualization, "-vis", "--visualization", "-no-vis",
+                  "--no-visualization",
+                  "Enable or disable GLVis visualization.");
+   args.Parse();
+   if (!args.Good())
+   {
+      args.PrintUsage(cout);
+      return 1;
+   }
+   args.PrintOptions(cout);
+
+   // 2. Enable hardware devices such as GPUs, and programming models such as
+   //    CUDA, OCCA, RAJA and OpenMP based on command line options.
+   Device device(device_config);
+   device.Print();
+
+   // 3. Read the mesh from the given mesh file. We can handle triangular,
+   //    quadrilateral, tetrahedral, hexahedral, surface and volume meshes with
+   //    the same code.
+   Mesh *mesh = new Mesh(mesh_file, 1, 1);
+   int dim = mesh->Dimension();
+
+   // 4. Refine the mesh to increase the resolution. In this example we do
+   //    'ref_levels' of uniform refinement. We choose 'ref_levels' to be the
+   //    largest number that gives a final mesh with no more than 10,000
+   //    elements.
+   {
+      if (ref_levels < 0)
+      {
+         ref_levels =
+            (int)floor(log(10000./mesh->GetNE())/log(2.)/dim);
+      }
+      for (int l = 0; l < ref_levels; l++)
+      {
+         mesh->UniformRefinement();
+      }
+   }
+
+   // 5. Define a finite element space on the mesh. Here we use the
+   //    Raviart-Thomas finite elements of the specified order.
+   FiniteElementCollection *hdiv_coll = nullptr;
+   FiniteElementCollection *l2_coll = nullptr;
+   NURBSExtension *NURBSext = nullptr;
+
+   if (mesh->NURBSext&& NURBS)
```

Comment:
`NURBS` is undefined, and this example does not compile.

Reply:
I think I had a mistake in the make file and the testcase was running ex3 instead of ex5.

Do not know what is going on...

Reply:
Fixed the NURBS flag. But still issues.

Reply:
Implemented the correct boundary element HDiv  FECollection. This makes it seem to work (compile + run).
Might need some double checking if things are actually correct.